34.md

```diff
@@ -0,0 +1,102 @@
+NIP-34
+======
+
+`git` stuff
+-----------
+
+`draft` `optional`
+
+This NIP defines all the ways code collaboration using and adjacent to [`git`](https://git-scm.com/) can be done using Nostr.
+
+## Repository announcements
+
+Git repositories are hosted in Git-enabled servers, but their existence can be announced using Nostr events, as well as their willingness to receive patches, bug reports and comments in general.
+
+```jsonc
+{
+  "kind": 30617,
+  "content": "",
+  "tags": [
+    ["d", "<repo-id>"],
+    ["name", "<human-readable project name>"],
+    ["description", "brief human-readable project description>"],
+    ["web", "<url for browsing>", ...], // a webpage url, if the git server being used provides such a thing
+    ["clone", "<url for git-cloning>", ...], // a url to be given to `git clone` so anyone can clone it
+    ["relays", "<relay-url>", ...] // relays that this repository will monitor for patches and issues
+  ]
+}
+```
+
+The tags `web`, `clone`, `relays` can have multiple values.
+
+Except `d`, all tags are optional.
+
+## Patches
+
+Patches can be sent by anyone to any repository. Patches to a specific repository SHOULD be sent to the relays specified in that repository's announcement event's `"relays"` tag. Patch events SHOULD include an `a` tag pointing to that repository's announcement address.
+
+```jsonc
+{
+  "kind": 1617,
+  "content": "<patch>", // contents of <git format-patch>
+  "tags": [
+    ["a", "30617:<base-repo-owner-pubkey>:<base-repo-id>"],
+    ["p", "<repository-owner>"],
+    ["p", "<other-user>"], // optionally send the patch to another user to bring it to their attention
```

Comment:
can say a default limit (e.g `20`’s `other-user`).
relays can have local policy saying they allow more.
or price them with `zaps` or other micro-payments.
otherwise amplification attack vector for relay.

Reply:
events that tag large numbers of pubkeys or events can be identified as spam either at the relay level or by clients.
I'm not sure there needs to be a default limit as other spam identification criteria can be used such as web of trust, reports, etc.